packages/headless/src/state/state-sections.ts

```diff
@@ -250,6 +251,13 @@ export interface ProductListingSection {
   productListing: ProductListingState;
 }
 
+export interface ProductListingSortSection {
+  /**
+   * The sort to use with the product listing query.
+   */
+  productListingSort: ProductListingSortState;
```

Comment:
Is the `productListing` prefix needed?

Reply:
Hmm, maybe not.

But then, I think instead of naming the whole thing `ProductListingSortState`, maybe it should be `SearchAPIV4SortState` ? :thinking:

Reply:
The naming of the interface is a really good point.

At the start of the project, my thinking was it would be helpful if the state names matched the api.
While there are some advantages to this, one draw-back I learned of is it makes it hard to stay in sync when the api design changes.

So, I think it's better to not mention the api or api version in the interface, because they might not hold when migrating apis.
I think the names in state need to be owned by Headless. (e.g. `SortState`, `SortCriteriaState`)

The api clients then act as a bridge, mapping the headless representation to the api request format (if they match, then all the better, the state can just be passed through).
The reducers map the api response format back to the headless representation.

**Summary**
I think naming the interface in a way that does not mention the api would be best. e.g. `SortState`, `SortCriteriaState`